src/engine/controls/bpmcontrol.cpp

```diff
@@ -596,7 +596,7 @@ bool BpmControl::getBeatContextNoLookup(
         mixxx::FramePos pNextBeat,
         mixxx::FrameDiff_t* dpBeatLength,
         double* dpBeatPercentage) {
-    if (pPrevBeat == mixxx::FramePos(-1) || pNextBeat == mixxx::FramePos(-1)) {
+    if (pPrevBeat == mixxx::kInvalidFramePos || pNextBeat == mixxx::kInvalidFramePos) {
```

Comment:
I thought we were going to use std::optional instead?

Reply:
That wouldn't play out well with these C style functions like `findPrevNextBeat` where pointers to `previousBeat` and `nextBeat` are passed to them. std::optional was only suited for functions with `FramePos` return type.